src/app/components/Search/Search.jsx

```diff
@@ -85,11 +92,27 @@ class Search extends React.Component {
       searchKeywords,
       page: '1',
     });
+    // Need to save a copy of the present value of this.state.field because
+    // it's going to be overwritten by this.props.field at least once before
+    // the ajax call returns, causing this.state.field to be mismatched with
+    // the actual selection, creating much confusion for the visitor..
+    const searchField = String(this.state.field);
 
     this.props.updateIsLoadingState(true);
 
     return new Promise((resolve, reject) => {
       ajaxCall(`${appConfig.baseUrl}/api?${apiQuery}`, (response) => {
+        // TODO Might it be helpful to have a "SearchQuery" model that
+        // stores all of the distinct properties that represent a search
+        // (i.e. keyword, filters, page, sort, search_scope, etc.)
+        // so that the store has only one thing to hold onto. The following
+        // series of Actions calls should be atomic.
+        // We'd maintain one instance of a SearchQuery in the Alt store,
+        // representing the "current" query associated with the current
+        // results. This Search component would own it's own instance of that
+        // model in state, reprenting the more transient state of the form
```

Comment:
`representing`

Reply:
I repent! thanks. fixed

Reply:
I was just reading through the good comments and caught it!